Assets/__Scripts/MapEditor/Mapping/PlacementControllers/PlacementController.cs

```diff
@@ -216,10 +216,16 @@ private void FinishDrag()
         if (!(isDraggingObject || isDraggingObjectAtTime)) return;
         //First, find and delete anything that's overlapping our dragged object.
         var selected = SelectionController.IsObjectSelected(draggedObjectData);
-        objectContainerCollection.RemoveConflictingObjects(new[] { draggedObjectData }, out List<BeatmapObject> conflicting);
+
+        // To delete properly we need to set the original time
+        float _time = draggedObjectData._time;
+        draggedObjectData._time = originalDraggedObjectData._time;
+        objectContainerCollection.DeleteObject(draggedObjectData, false, false);
```

Comment:
Couldn't we just delete `originalDraggedObjectData`?

Reply:
No, the draggedObjectData points to an object that is inside LoadedObjects, it needs to have the correct _time value when it's removed.

You could _call_ this function with originalDraggedObjectData but you still have to make the change to draggedObjectData or it won't be removed.